sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -270,46 +268,69 @@ AZ_INLINE az_result _az_iot_provisioning_client_parse_payload_error_code(
 {
-  if (tm->token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  if (jp->token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
   {
     return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
   }
 
   bool found_assigned_hub = false;
   bool found_device_id = false;
 
-  while ((!(found_device_id && found_assigned_hub))
-         && az_succeeded(az_json_parser_parse_token_member(jp, tm)))
+  AZ_RETURN_IF_FAILED(az_json_parser_next_token(jp));
+
+  while (jp->token.kind != AZ_JSON_TOKEN_END_OBJECT)
   {
-    if (az_span_is_content_equal(AZ_SPAN_FROM_STR("assignedHub"), tm->name))
+    if (az_json_token_is_text_equal(&jp->token, AZ_SPAN_FROM_STR("assignedHub")))
     {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(jp));
+      if (jp->token.kind != AZ_JSON_TOKEN_STRING)
+      {
+        return AZ_ERROR_ITEM_NOT_FOUND;
+      }
+      out_state->assigned_hub_hostname = jp->token.slice;
       found_assigned_hub = true;
-      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm->token, &out_state->assigned_hub_hostname));
     }
-    else if (az_span_is_content_equal(AZ_SPAN_FROM_STR("deviceId"), tm->name))
+    else if (az_json_token_is_text_equal(&jp->token, AZ_SPAN_FROM_STR("deviceId")))
     {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(jp));
+      if (jp->token.kind != AZ_JSON_TOKEN_STRING)
+      {
+        return AZ_ERROR_ITEM_NOT_FOUND;
+      }
+      out_state->device_id = jp->token.slice;
       found_device_id = true;
-      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm->token, &out_state->device_id));
     }
-    else if (az_span_is_content_equal(AZ_SPAN_FROM_STR("errorMessage"), tm->name))
+    else if (az_json_token_is_text_equal(&jp->token, AZ_SPAN_FROM_STR("errorMessage")))
     {
-      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm->token, &out_state->error_message));
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(jp));
+      if (jp->token.kind != AZ_JSON_TOKEN_STRING)
+      {
+        return AZ_ERROR_ITEM_NOT_FOUND;
+      }
+      out_state->error_message = jp->token.slice;
     }
-    else if (az_span_is_content_equal(AZ_SPAN_FROM_STR("lastUpdatedDateTimeUtc"), tm->name))
+    else if (az_json_token_is_text_equal(&jp->token, AZ_SPAN_FROM_STR("lastUpdatedDateTimeUtc")))
     {
-      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm->token, &out_state->error_timestamp));
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(jp));
+      if (jp->token.kind != AZ_JSON_TOKEN_STRING)
+      {
+        return AZ_ERROR_ITEM_NOT_FOUND;
+      }
+      out_state->error_timestamp = jp->token.slice;
     }
-    else if (tm->token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT)
+    else if (az_succeeded(_az_iot_provisioning_client_parse_payload_error_code(jp, out_state)))
     {
-      AZ_RETURN_IF_FAILED(az_json_parser_skip_children(jp, tm->token));
+      // Do nothing
     }
     else
     {
-      (void)_az_iot_provisioning_client_parse_payload_error_code(tm, out_state);
+      // ignore other tokens
```

Comment:
I don't believe this logic really change. The parser was still being advanced before, like this:
```C
else if (tm->token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT)
{
   AZ_RETURN_IF_FAILED(az_json_parser_skip_children(jp, tm->token));
}
```

The reason why it got moved to the else block rather than a check on `AZ_JSON_TOKEN_BEGIN_OBJECT` is because the parser no longer returns key/value pair `az_json_token_member`, but rather moves one `az_json_token` at a time.